narwhals/_arrow/utils.py

```diff
@@ -246,3 +250,32 @@ def cast_for_truediv(arrow_array: Any, pa_object: Any) -> tuple[Any, Any]:
         )
 
     return arrow_array, pa_object
+
+
+def validate_shape(
+    series: list[ArrowSeries], backend_version: tuple[int, ...]
+) -> list[Any]:
+    pa = get_pyarrow()
+    lengths = [len(s) for s in series]
+    max_length = max(lengths)
+    fast_path = all(_len == max_length for _len in lengths)
+
+    if fast_path:
+        return [s._native_series for s in series]
+
+    reshaped = []
+
+    for s, length in zip(series, lengths):
+        if max_length > 1 and length == 1:
+            value = s._native_series[0]
+
+            if backend_version < (13,) and hasattr(value, "as_py"):  # pragma: no cover
+                value = value.as_py()
+
+            reshaped.append(
+                pa.chunked_array([[value] * max_length], type=s._native_series.type)
```

Comment:
Should we create the array with numpy instead of a python list?

Reply:
i think it doesn't matter, and what you've done should be faster

Reply:
I somehow am afraid that time is similar, but `np.full(max_length, value)` is more memory efficient, especially if we had a way to map arrow to numpy datatypes

Reply:
sure let's do that